code/modules/shuttle/computer.dm

```diff
@@ -24,13 +24,19 @@
 	var/list/options = params2list(possible_destinations)
 	var/obj/docking_port/mobile/M = SSshuttle.getShuttle(shuttleId)
 	var/dat = "Status: [M ? M.getStatusText() : "*Missing*"]<br><br>"
-	if(M)
+	if(istype(src, /obj/machinery/computer/shuttle/fob) && SSstarmap.in_transit || SSstarmap.in_transit_planet)
+		dat += "<B>Main ship is currently in transit.</B><br>"
+	else if(M && M.mode == SHUTTLE_TRANSIT)
+		dat += "<B>The shuttle is currently in transit.</B><br>"

```

Comment:
this is a gross breach of oop practices. Use inheritance

Reply:
Resolved, hopefully.